eng/doc/DeveloperGuide.md

```diff
@@ -0,0 +1,119 @@
+# Developer Guide
+
+This document is a guide for developers who want to contribute to the Microsoft Go repository.
+It explains how to build the repository, how to work with the Go submodule, and how to use the different tools that help maintain the repository.
+
+This guide is primarily intended for developers working for the Go team at Microsoft, but it can also be useful for external contributors.
+
+## Setting up the repository
+
+### Step 0: Contributor License Agreement
+
+Most contributions require you to agree to a Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us the rights to use your contribution.
+For details, visit https://cla.opensource.microsoft.com.
+
+### Step 1: Install a Go toolchain
+
+A preexisting Go toolchain is required to bootstrap the build process.
+You can use your system's package manager to install Go, or you can download it from the [official Go website](https://golang.org/dl/).
+The only requirement is that the Go version is high enough for the bootstrap process.
+If the version is too low, the bootstrap process will fail and ask you to install a newer version.
+
+This repository implements some scripts (provided by `eng/run.ps1`) to facilitate installing the correct bootstrapping Go version and also to build the Go toolchain from source, see the [`eng` Readme](../eng/README.md) for more information.
+It is recommended that you get familiar with both the upstream Go build process and the scripts provided in this repository.
+
+### Step 2: Install git and the git-go-patch command
```

Comment:
Multiple Step 2's here, I suggest removing the numbers altogether as it makes further editing easier

Reply:
Done!